cmake/BLTThirdPartyConfigFlags.cmake.in

```diff
@@ -1,18 +1,46 @@
 if ("@ENABLE_CUDA@")
-    set(BLT_ENABLE_CUDA     "@ENABLE_CUDA@")
+    set(BLT_ENABLE_CUDA           "@ENABLE_CUDA@")
 endif()
 
 if ("@ENABLE_HIP@")
-    set(BLT_ENABLE_HIP      "@ENABLE_HIP@")
+    set(BLT_ENABLE_HIP            "@ENABLE_HIP@")
 endif()
 
 if ("@ENABLE_MPI@")
-    set(BLT_ENABLE_MPI      "@ENABLE_MPI@")
+    set(BLT_ENABLE_MPI            "@ENABLE_MPI@")
 endif()
 
 if ("@ENABLE_OPENMP@")
-    set(BLT_ENABLE_OPENMP   "@ENABLE_OPENMP@")
+    set(BLT_ENABLE_OPENMP         "@ENABLE_OPENMP@")
 endif()
 
-set(BLT_ENABLE_FIND_MPI     "@ENABLE_FIND_MPI@")
-set(BLT_ENABLE_CLANG_CUDA   "@ENABLE_CLANG_CUDA")
+set(BLT_ENABLE_FIND_MPI           "@ENABLE_FIND_MPI@")
+
+# User configurable CUDA options needed by downstream.  Normally, exporting
+# these targets would pass these flags to downstream targets, but this
+# must be done manually if targets aren't exported.
+if ("@ENABLE_CLANG_CUDA@" AND NOT DEFINED ENABLE_CLANG_CUDA)
+    set(BLT_ENABLE_CLANG_CUDA     "@ENABLE_CLANG_CUDA@")
+# If the user specifies this variable, ignore the inherited flag.
+elseif (DEFINED ENABLE_CLANG_CUDA)
+    set(BLT_ENABLE_CLANG_CUDA     ${ENABLE_CLANG_CUDA})
+endif()
+
+if ("@BLT_CLANG_CUDA_ARCH@" AND NOT DEFINED CLANG_CUDA_ARCH)
+    set(BLT_CLANG_CUDA_ARCH       "@BLT_CLANG_CUDA_ARCH@")
+elseif (DEFINED CLANG_CUDA_ARCH)
+    set(BLT_CLANG_CUDA_ARCH       ${CLANG_CUDA_ARCH})
+endif()
+
+if ("@CUDA_TOOLKIT_ROOT_DIR@" AND NOT DEFINED CUDA_TOOLKIT_ROOT_DIR)
+    set(BLT_CUDA_TOOLKIT_ROOT_DIR "@CUDA_TOOLKIT_ROOT_DIR@")
+elseif (DEFINED CUDA_TOOLKIT_ROOT_DIR)
+    set(BLT_CUDA_TOOLKIT_ROOT_DIR ${CUDA_TOOLKIT_ROOT_DIR})
+endif()
+
+# User configurable HIP options needed by downstream
+if ("@ROCM_PATH@" AND NOT DEFINED ROCM_PATH)
```

Comment:
`ROCM_PATH` isn't ours and probably shouldn't be prefixed with `BLT_`.

https://docs.amd.com/projects/HIP/en/latest/developer_guide/contributing.html#make-tips

Reply:
Same with `CUDA_TOOLKIT_ROOT_DIR`. We should leave the setting the variable if not set by the downstream user. but remove the prefix imo.  Thoughts?

Reply:
I think setting is good in the case that the downstream project is not using the TPLs, only the installed library. In the case where the downstream project is using a TPL, it should have already been configured and this won't be evaluated again.